RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -167,74 +205,88 @@ void HiEvtPlaneFlatProducer::produce(edm::Event& iEvent, const edm::EventSetup&
       }
     }
   }
-
+  //
+  //Get flattening parameter file.
+  //
   if (hirpWatcher.check(iSetup)) {
     edm::ESHandle<RPFlatParams> flatparmsDB_;
     iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
     LoadEPDB db(flatparmsDB_, flat);
-    if (!db.IsSuccess())
-      return;
-  }
+  }  //rp record change
+
   //
   //Get Centrality
   //
-
-  int bin = iEvent.get(centralityBinToken) / CentBinCompression_;
-
-  if (Noffmin_ >= 0) {
-    int nOff = iEvent.get(centralityToken).Ntracks();
-    if ((nOff < Noffmin_) or (nOff >= Noffmax_)) {
-      return;
-    }
-  }
+  int bin = 0;
+  int cbin = 0;
+  edm::Handle<int> cbin_;
+  iEvent.getByToken(centralityBinToken, cbin_);
+  cbin = *cbin_;
+  bin = cbin / CentBinCompression_;
+  //double centval = cbin*100./(double) nCentBins_;
   //
   //Get Vertex
   //
-  int vs_sell;  // vertex collection size
-  float vzr_sell;
-  auto const& vertices3 = iEvent.get(vertexToken);
-  vs_sell = vertices3.size();
-  if (vs_sell > 0) {
-    vzr_sell = vertices3.begin()->z();
-  } else
-    vzr_sell = -999.9;
+  edm::Handle<reco::VertexCollection> vertices;
+  iEvent.getByToken(vertexToken, vertices);
+
+  //best vertex
+  double bestvz = -999.9, bestvx = -999.9, bestvy = -999.9;
+  const reco::Vertex& vtx = (*vertices)[0];
+  bestvz = vtx.z();
+  bestvx = vtx.x();
+  bestvy = vtx.y();
+  math::XYZPoint bestvtx(bestvx, bestvy, bestvz);
+  if (bestvz < minvtx_ || bestvz > maxvtx_)
+    return;
 
   //
   //Get Event Planes
   //
 
+  edm::Handle<reco::EvtPlaneCollection> evtPlanes_;
+  iEvent.getByToken(inputPlanesToken, evtPlanes_);
+
+  if (!evtPlanes_.isValid()) {
+    return;
```

Comment:
producer modules are not allowed to return without making a product at least in reco context.
Also, a silent return should be really well justified, it seems like it's a quiet way to skip configuration problems, which is not appropriate for modules to be used in production